squidasm/sim/stack/netstack.py

```diff
@@ -325,42 +318,20 @@ def handle_create_ck_request(
 
             # Put the request to the EGP.
             self._logger.info(f"putting CK request for pair {pair_index}")
-            self._egp.put(request)
+            current_egp = self._egp[req.remote_node_id]
+            current_egp.put(request)
 
             # Wait for a signal from the EGP.
             self._logger.info(f"waiting for result for pair {pair_index}")
             yield self.await_signal(
-                sender=self._egp, signal_label=ResCreateAndKeep.__name__
+                sender=current_egp, signal_label=ResCreateAndKeep.__name__
             )
             # Get the EGP's result.
-            result: ResCreateAndKeep = self._egp.get_signal_result(
+            result: ResCreateAndKeep = current_egp.get_signal_result(
                 ResCreateAndKeep.__name__, receiver=self
             )
             self._logger.info(f"got result for pair {pair_index}: {result}")
 
-            # This code is commented out for a hotfix as it was found that heralded link did not return
```

Comment:
I see this code is removed completely now. Is this something that is planned to be put back in?

Reply:
No, from what I gather, it seems that NetQASM does the bell state correction. Doing any bell state correction here would give incorrect results for any model with non-default bellIndex